### DISCUSSION TITLE
How to fix empty maps after update?

### DISCUSSION THREAD

**Discussion by LordXerus** [ppy/osu#16674] [Q&A]
I updated to version 128 from 118 today. It seemed to have performed some kind of migration when the game started.

After updating, some of the maps seems to have lost all their hit objects. Re-importing the map can restore the hit objects, but it wipes the score as well. Is there a way to rollback the client or database?

**Comment by peppy** [comment]
Could you please open your osu! folder (there's a button towards the top of settings to do this) and zip the `backups` folder, then attach to this thread? I'll see what I can do to recover your data.

Also could you explain what you mean by "lost all their hit objects"? Does gameplay still load?

**Reply by LordXerus** [reply]
One of the difficulties will be 0* because as far as I can tell, the map just has no hit objects. Upon loading, it will complain that the map has no hit objects. After closing the notifications, it will load back to the selection screen.

So far I have observed 2 instances of maps dissappearing:
- Once inside the song _Ticking_ (as shown below). This one I don't play much.
- _Masquerade_, this one I do play and lost scores because re-import.

Another things is that I have low disk space (around 2GB or so), which might have caused the failure. Is there a way to move the songs folder?

Screenshots of 0* empty level, [backups.zip](https://github.com/ppy/osu/files/7956313/backups.zip) attached

![image](https://user-images.githubusercontent.com/34696211/151499815-383e274a-3f2f-40d9-b9ec-fbbed7881336.png)
![image](https://user-images.githubusercontent.com/34696211/151499754-bbdcc267-8b5e-4a7e-94b9-c6d72c1b9d08.png)




[backups.zip](https://github.com/ppy/osu/files/7956313/backups.zip)

**Reply by peppy** [reply]
Hmm, this is super weird. It almost looks unrelated to the database migration (since it doesn't touch the beatmap files on disk). Are you sure those beatmaps were working quite recently?

I'll check on your database for sanity, but in the mean time: you can move the songs folder by opening settings and searching for "change". There should be a "Change folder location..." button that will let you move it to another drive.

**Reply by peppy** [reply]
Just to confirm, is there no third file in your backups folder? If not, can you also please zip and send your `client.realm` file (in the main folder)?

**Reply by LordXerus** [reply]
I only see two files.
I'm pretty sure _Masquerade_ is fine. I think I played with it yesterday.
I haven't played _Ticking_ except once on multiplayer.

[client.zip](https://github.com/ppy/osu/files/7956626/client.zip)

**Reply by peppy** [reply]
Hmm. From a database perspective everything looks correct. I can't see any reason why the issue you are experiencing could happen.

Could you try opening one of the affected beatmaps in the editor and see what happens? If it loads okay, can you choose "File > Export" and attach the osz file it generates?

**Reply by LordXerus** [reply]
Actually, I think the blank map in _Ticking (Graveyard)_ in intentional. It's the same thing on the beatmap listing as well.

However, the _Masquerade (Ranked)_ case was not. The Insane map was erased. Unfortunately, I already reimported the map, so I don't think there's anything interesting here.

I haven't found any other maps that did this, however. Is there a way to find maps with 0* difficulty?

**Reply by peppy** [reply]
`star=0` should work

**Reply by bdach** [reply]
The song select text filters don't update dynamically though, currently the value that is used comes from the initial DB import. So if the DB contains a nonzero difficulty and migrated properly then `star=0` won't work.

**Reply by LordXerus** [reply]
Sorry, GitHub didn't let me reply yesterday for some reason.

I didn't find any more empty maps, but is it possible to get back my scores for Masquerade?

**Reply by peppy** [reply]
you can restore the database I sent you, it should have all scores loaded.

**Reply by LordXerus** [reply]
Sorry, slightly confused about procedure. Which comment did you send the database in? (quote the comment) and how to restore?
